LDAR_Sim/src/economics/cost_mitigation.py

```diff
@@ -0,0 +1,174 @@
+                          ['GWP_CH4'], 'carbon_price_tonnesCO2e': df['p_params']['economics']
+                          ['carbon_price_tonnesCO2e'], 'social_cost_CH4_tonnes': df['p_params']
+                          ['economics']['social_cost_CH4_tonnes'],
+                          'cost_CCUS': df['p_params']['economics']['cost_CCUS'],
+                          'cost_low_bleed_pneu_tCO2e': df['p_params']['economics']
+                          ['cost_low_bleed_pneu_tCO2e'],
+                          'total_program_cost': df['timeseries']['total_daily_cost'].sum()}
+                         for df in simulation_dfs]
+    economics_outputs_df = pd.DataFrame(economics_outputs)
+
+    # Need to verify if P_base was input. If not, economics module will not work.
+    if 'P_ref' in economics_outputs_df.values:
+        print("P_ref input. Economics functionality running.")
+    else:
+        raise Exception("No P_ref input. Cannot run economics.")
+
+    # Calculate the mean emissions and costs accross all simulations, convert units.
+    economics_df = economics_outputs_df.groupby(by='program_name').mean()
+
+    economics_df['total_emissions_mcf'] = (((economics_df['total_program_emissions_kg']
+                                             / 0.678) * 35.3147) / 1000)
+    economics_df['simulation_avg_emissions_tonnesCO2e'] = ((economics_df
+                                                            ['total_program_emissions_kg'] / 1000) *
+                                                           economics_df['GWP_CH4'])
+
+    # Find the simulation average emissions from the base program. Subtract from programs.
+    P_ref_value = economics_df.loc['P_ref', 'total_emissions_mcf']
+    economics_df['difference_ref_mcf'] = (economics_df['total_emissions_mcf'] - P_ref_value)
+
+    # Take difference in emissions and multiply by sale price of natural gas.
+    # Check to make sure Alt-Program achieves reductions over Base Program first.
+    for row in economics_df['difference_ref_mcf']:
+        if row <= 0:
+            economics_df['value_gas_sold'] = (
+                abs(economics_df['difference_ref_mcf']) * economics_df['sale_price_natgas'])
+        else:
+            economics_df['value_gas_sold'] = 0
+
+    # Find difference from baseline in tonnes CO2e. Use value for cost/mitigation ratio.
+    # Need to verify that Alt-Program achieves reductions over Base Program too.
+    for row in economics_df['difference_ref_mcf']:
+        if row <= 0:
+            economics_df['dif_ref_tonnesCO2e'] = (((((abs(economics_df
+                                                          ['difference_ref_mcf']) * 1000)
+                                                     / 35.3147) * 0.678) / 1000) *
+                                                  economics_df['GWP_CH4'])
+        else:
+            economics_df['dif_ref_tonnesCO2e'] = 0
+
+    economics_df['cost_mitigation_ratio'] = np.divide(economics_df['total_program_cost'],
```

Comment:
```
economics_df['cost_mitigation_ratio'] = np.divide(
     economics_df['total_program_cost'],
     economics_df['dif_ref_tonnesCO2e'],
     out=np.zeros_like(
        economics_df['total_program_cost']),
        where=economics_df['dif_ref_tonnesCO2e'] != 0)
```

Reply:
Not a big deal but PEP 8 Would tell you if you have nested methods/functions (divide and zeros_like) to line break after the functions opening brackets.